File: editorial_ch/chap5-9/prob5-9-3.py
# 輸入
# 注意 A 為 (結束時間, 開始時間) [為了以結束時間的遞增順序排序]
N = int(input())
A = []
for i in range(N):
	a, b = map(int, input().split())
	A.append([b, a])

# 排序
A.sort()

# 持續選擇結束時間最早的電影
CurrentTime = 0
Answer = 0
for i in range(N):
	if CurrentTime <= A[i][1]:
		CurrentTime = A[i][0]
		Answer += 1

# 輸出
print(Answer)

File: editorial_ch/chap5-9/prob5-9-3.java
import java.util.*;

class Main {
	public static void main(String[] args) {
		// 輸入
		Scanner sc = new Scanner(System.in);
		int N = sc.nextInt();
		int[][] A = new int[N][2];
		for (int i = 1; i <= N; i++) {
			A[i - 1][0] = sc.nextInt();
			A[i - 1][1] = sc.nextInt();
		}
		
		// 將陣列 A 以結束時間（A[x][1]）的遞增順序排序
		Arrays.sort(A, (a1, a2) -> Integer.compare(a1[1], a2[1]));
		
		// 持續選擇結束時間最早的電影
		int CurrentTime = 0;
		int Answer = 0;
		for (int i = 1; i <= N; i++) {
			if (CurrentTime <= A[i - 1][0]) {
				CurrentTime = A[i - 1][1];
				Answer += 1;
			}
		}
		
		// 輸出
		System.out.println(Answer);
	}
}